src/helpers/WLClasses.cpp

```diff
@@ -8,16 +8,23 @@ SLayerSurface::SLayerSurface() {
 }
 
 void SLayerSurface::applyRules() {
-    noAnimations = false;
-    forceBlur    = false;
-    ignoreZero   = false;
+    noAnimations     = false;
+    forceBlur        = false;
+    ignoreAlpha      = false;
+    ignoreAlphaValue = 0.f;
 
     for (auto& rule : g_pConfigManager->getMatchingRules(this)) {
         if (rule.rule == "noanim")
             noAnimations = true;
         else if (rule.rule == "blur")
             forceBlur = true;
-        else if (rule.rule == "ignorezero")
-            ignoreZero = true;
+        else if (rule.rule.find("ignorealpha") == 0 || rule.rule.find("ignorezero") == 0) {
+            const std::string VALUE = rule.rule.substr(rule.rule.find_first_of(' ') + 1);
+            try {
+                ignoreAlpha = true;
+                if (VALUE.size() != 0)
```

Comment:
value.empty()

also, check if it has a space at all (or is it npos) and ignore if not

Reply:
yessir
https://github.com/hyprwm/Hyprland/pull/2477/commits/695c02f1193d682ffd5b5f3af109af8d66298b0d
https://github.com/hyprwm/Hyprland/pull/2477/commits/a7c34572a0edc2b8e3807060fce34bfafa6748ae